Scripts/RemediationScripts/Remediate-EnableImmutabilitySettingForBackupVault.ps1

```diff
@@ -0,0 +1,541 @@
+    }
+  
+    Write-Host "[Step 3 of 4] Back up Backup Vault(s) details..."
+    Write-Host $([Constants]::SingleDashLine)
+ 
+    if ([String]::IsNullOrWhiteSpace($FilePath)) {
+        # Backing up Backup Vault(s) details.
+        $backupFile = "$($backupFolderPath)\BackupVaultDetailsBackUp.csv"
+        $backupVaultsWithoutImmutability | Export-CSV -Path $backupFile -NoTypeInformation
+        Write-Host "Backup Vault(s) details have been backed up to [$($backupFile)]" -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else {
+        Write-Host "Skipped as -FilePath is provided" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+ 
+    if (-not $DryRun) {
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Enable and lock immutability on Backup Vault(s) in the Subscription..." 
+        Write-Host $([Constants]::SingleDashLine)
+         
+        if (-not $Force) {
+            Write-Host "Do you want to enable and lock immutability on Backup Vault(s) in the Subscription? " -ForegroundColor $([Constants]::MessageType.Warning)
+            Write-Host "Note: This action is irreversible. Once locked, immutability cannot be disabled." -ForegroundColor $([Constants]::MessageType.Warning)
+            $userInput = Read-Host -Prompt "(Y|N)"
+            if ($userInput -ne "Y") {
+                Write-Host "Immutability will not be enabled and locked on Backup Vault(s) in the Subscription. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+                Write-Host $([Constants]::DoubleDashLine)    
+                return
+            }
+        }
+        else {
+            Write-Host "'Force' flag is provided. Proceeding to enable and lock immutability on Backup Vault(s) without any prompts..." -ForegroundColor $([Constants]::MessageType.Warning)
+        }
+ 
+        # List for storing remediated Backup Vaults(s)
+        $backupVaultsRemediated = @()
+ 
+        # List for storing skipped Backup Vaults(s)
+        $backupVaultsSkipped = @()
+ 
+        # Enable and lock immutability on each Backup Vault
+        # Loop through the list of Backup Vault(s) which needs to be remediated.
+        $backupVaultsWithoutImmutability | ForEach-Object {
+            $backupVault = $_
+            try {
+                if (-not $_.ImmutabilityState) {
+                    Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -ImmutabilityState Disabled -ErrorAction Stop                   
+                }
+                $backupVaultResource = Update-AzDataProtectionBackupVault -ResourceGroupName $_.ResourceGroupName -VaultName $_.ResourceName -ImmutabilityState Locked -ErrorAction Stop   
+ 
+                if ($backupVaultResource.ImmutabilityState.ToString() -ieq "LOCKED") {
```

Comment:
can we check whether ImmutablityState is null or not?

Reply:
After updating the immutability state. It won't be null. If the above Update-AzDataProtectionBackupVault failed to update ImmutabilityState it will throw an error which is handled by try catch.